Source/ObjectiveDropboxOfficial/Shared/Handwritten/Networking/DBTasksImpl.m

```diff
@@ -27,13 +29,30 @@ - (instancetype)initWithTask:(NSURLSessionDataTask *)task
   return self;
 }
 
+- (NSURLSessionTask *)currentTask {
+  return _dataTask;
+}
+
+- (void)restart {
+  NSURLRequest *request = [[self currentTask].originalRequest copy];
+
+  [[NSOperationQueue mainQueue] addOperationWithBlock:^{
+    NSURLSessionDataTask *task = [_session dataTaskWithRequest:request];
+    self.retryCount += 1;
+    _dataTask = task;
+    [self setResponseBlock:[_responseBlock copy] queue:_queue];
```

Comment:
In these blocks, you don't need `copy`. That's the responsibility of the callee. Also, in ARC, you generally don't need to `copy` blocks as it's done automatically (if necessary) where objects are normally `retain`ed.